### ISSUE TITLE
Get-CertificateTemplate reports wrong result for AutoenrollmentAllowed

### ISSUE THREAD

**Issue by gnugnug** [PKISolutions/PSPKI#218]
I was under the impression that the column _AutoenrollmentAllowed_ of the Cmdlet Get-CertificateTemplate reports the same value as the column _Windows XP Autoenrollment_ in certtmpl.msc.
However when looking at the values they differ. For some templates certtmpl.msc. shows "Allowed" while Get-CertificateTemplate shows "False", for other templates it's the other way around. I believe the values from certtmpl.msc are correct, because Get-CertificateTemplate sometimes reports _AutoenrollmentAllowed=true_ for offline templates, which shouldn't be possible.
Or is _AutoenrollmentAllowed_ something completely different?

**Comment by Crypt32** [comment]
I don't know how exactly Autoenrollment column works in certtmpl.msc, I wasn't able to find any references. Though, cert template contains `Autoenrollment` flag, which is set by certtmpl.msc:
```
public Boolean AutoenrollmentAllowed => SchemaVersion > 1 && (_template.Flags & CertificateTemplateFlags.Autoenrollment) != 0;
```
So my property returns when template schema version is >1 and template flags assert Autoenrollment bit.

> sometimes reports AutoenrollmentAllowed=true for offline templates, which shouldn't be possible

in fact, it is not quite correct statement. Offline templates can be used by autoenrollment -- in renewal scenarios. Initial autoenrollment is not possible, but autoenrollment can renew templates based on offline templates if renewal accepts existing certificate as renewal requirement.

**Comment by Crypt32** [comment]
> Does _template.Flags_ in your code example reference the AD attribute _flags_ (which I guess translates to GeneralFlags in PSPKI)

yes, `_template.Flags` maps to `flags` DS attribute and translates to `GeneralFlags` in PSPKI.

It seems that `Autoenrollment` flag exist in two distinct enumerations:
1. https://github.com/PKISolutions/pkix.net/blob/32832132ce6eb782d339bd48aa0c12d7fa0f4adc/src/SysadminsLV.PKI.Win/CertificateTemplates/CertificateTemplateEnrollmentFlags.cs#L43
1. https://github.com/PKISolutions/pkix.net/blob/32832132ce6eb782d339bd48aa0c12d7fa0f4adc/src/SysadminsLV.PKI.Win/CertificateTemplates/CertificateTemplateFlags.cs#L37

and they are not synchronized or have different meaning and I (probably) incorrectly use `CertificateTemplateFlags.Autoenrollment` instead of correct `CertificateTemplateEnrollmentFlags.Autoenrollment`. Just to clarify that GeneralFlags and EnrollmentFlags are different enumerations.

> Where does EnrolleeSuppliesSubject in the flags of Template 1 come from? This is an online template

according to flags, it doesn't seem like this template is online (builds subject from AD).

**Comment by gnugnug** [comment]
I compared all our templates and the column "_Autoenrollment_" in certtmpl.msc shows "_Allowed_" for all templates where `msPKI-Enrollment-Flag & 0x020 == true`. On the other hand PSPKI uses the attribute "_flags_" for the comparison, which leads to different results. So the confusion has been cleared up, thanks for your help!
Maybe you can change this in a future version to use _msPKI-Enrollment-Flag_ for the AutoEnrollent setting, so that the output of both programs matches. The AutoEnrollment bit in the _flags_ attribute doesn't seem to be necessarary for the successful autoenrollment of certificates.

Independant of this PSPKI always translates a _flags_ value with the least bit set to "EnrolleeSuppliesSubject", regardless of the settings on the Subject tab of the template, because of https://github.com/PKISolutions/pkix.net/blob/32832132ce6eb782d339bd48aa0c12d7fa0f4adc/src/SysadminsLV.PKI.Win/CertificateTemplates/CertificateTemplateFlags.cs#L17 
Does 0x1 really mean EnrolleeSuppliesSubject? I haven't found this in the documention.

**Comment by Crypt32** [comment]
> On the other hand PSPKI uses the attribute "_flags_" for the comparison, which leads to different results.

then it is a bug in my code.
